### PR TITLE
Fix ItemData Serialization

### PR DESCRIPTION
**Pull request by APickledWalrus**
### Description
This PR fixes the ItemData serialization issues (e.g. assertion error for `diamond sword named "BOB"`) by serializing BlockValues stored on the ItemData. This is not currently done, which means recreated ItemData's are missing data.

We should probably port this to dev/2.6 - proper serialization will need to be done there too.

This is a breaking change. Because we are updating the ItemData serialization method, it will **NOT** be possible to downgrade your Skript version (to one without this fix) after using a Skript version containing this fix. Well, it will be possible, but variables will fail to load.

---
**Target Minecraft Versions:** Any
**Requirements:** N/A
**Related Issues:** I don't think any

### CODE REVIEW

**Review by TheLimeGlass — APPROVED** [review]

File: src/main/java/ch/njol/skript/bukkitutil/block/NewBlockCompat.java [hunk 1]
```diff
@@ -104,7 +116,26 @@ public MatchQuality match(BlockValues other) {
 				return MatchQuality.DIFFERENT;
 			}
 		}
-		
+
+		@Override
+		public Fields serialize() {
+			Fields fields = new Fields();
+			fields.putObject("data", data.getAsString());
+			fields.putPrimitive("isDefault", isDefault);
```
Comment (TPGamesNL) [inline src/main/java/ch/njol/skript/bukkitutil/block/NewBlockCompat.java:124]: Might be a good idea to assert / throw an exception in the constructor of NewBlockValues if `type` doesn't match the Material of `data`

**Review by TPGamesNL — COMMENTED** [review]
Make sure you test this on legacy versions as well, maybe it'd be a good idea to even create the PR for that on the dev/2.6 branch already. This will require a more elaborate solution than just a cherry pick; dev/2.6 has MagicBlockValues.

**Comment by APickledWalrus** [comment]
> Make sure you test this on legacy versions as well, maybe it'd be a good idea to even create the PR for that on the dev/2.6 branch already. This will require a more elaborate solution than just a cherry pick; dev/2.6 has MagicBlockValues.

Legacy versions will need their own PR, which I plan to make after this one is merged :)

**Review by TPGamesNL — APPROVED** [review]

**Review by AyhamAl-Ali — APPROVED** [review]

**Comment by APickledWalrus** [comment]
I've integrated #5105 into this PR. This avoids usage of Material registry completely and just replaced it with the direct calls it would be making. Build should pass, but needs tested with a bigger set of item variables.

**Review by TheLimeGlass — APPROVED** [review]

**Pull request merged**